rls-analysis/src/lowering.rs

```diff
@@ -242,10 +244,30 @@ impl<'a> CrateReader<'a> {
                     ve.insert(Ref::Id(def_id));
                 }
             }
-            analysis.ref_spans.entry(def_id).or_insert_with(|| vec![]).push(span);
+            analysis.ref_spans.entry(def_id).or_insert_with(Vec::new).push(span.clone());
+
+            #[cfg(feature = "idents")]
+            {
+                Self::record_ident(analysis, &span, def_id, IdentKind::Ref);
+            }
         }
     }
 
+    #[cfg(feature = "idents")]
+    fn record_ident(analysis: &mut PerCrateAnalysis, span: &Span, id: Id, kind: IdentKind) {
+        let row_start = span.range.row_start;
+        let col_start = span.range.col_start;
+        let col_end = span.range.col_end;
+        analysis
+            .idents
+            .entry(span.file.clone())
+            .or_insert_with(IdentsByLine::new)
+            .entry(row_start)
+            .or_insert_with(IdentsByColumn::new)
+            .entry(col_start)
+            .or_insert_with(|| IdentBound::new(col_end, id, kind));
```

Comment:
If I understand this correctly acts like a 2d segment tree for the spans?

Reply:
pretty much